source/module_hamilt_lcao/module_deltaspin/spin_constrain.cpp

```diff
@@ -537,8 +537,15 @@ void SpinConstrain<FPTYPE, Device>::print_Mi(bool print)
     {
         for (int iat = 0; iat < nat; ++iat)
         {
-            std::cout << "Total Magnetism on atom: " << iat << " " << std::setprecision(16) << " (" << Mi_[iat].x
-                      << ", " << Mi_[iat].y << ", " << Mi_[iat].z << ")" << std::endl;
+            if (this->nspin_ == 2)
+            {
+                std::cout << "Total Magnetism on atom: " << iat << " " << std::setprecision(10) << " (" << Mi_[iat].z << ")" << std::endl;
```

Comment:
I have two questions here:
1. why do you output these in screen rather than running file?
2. If delta spin is not used, can ABACUS print this atomic magnetism infomation as the same format?

Reply:
1. Output on screen is better for users to check the convergence process of atomic magnetic moments during scf steps.
2. No. This is an intrinsic drawback of ABACUS that needs attention in future.